zerver/lib/actions.py

```diff
@@ -446,6 +448,41 @@ def notify_created_user(user_profile: UserProfile) -> None:
     event = dict(type="realm_user", op="add", person=person)  # type: Dict[str, Any]
     send_event(user_profile.realm, event, active_user_ids(user_profile.realm_id))
 
+def notify_reactivated_user(user_profile: UserProfile) -> None:
+    user_data = get_raw_user_data(user_profile.realm,
+                                  acting_user=user_profile,
+                                  client_gravatar=False,
+                                  target_user=user_profile)
+    person = user_data[user_profile.id]
+    event = dict(type="realm_user", op="add", person=person)  # type: Dict[str, Any]
+    send_event(user_profile.realm, event, active_user_ids(user_profile.realm_id))
+
+    presence_dict = get_presence_for_user(user_profile)
+    if presence_dict:
+        event = dict(type="presence", email=user_profile.email,
+                     user_id=user_profile.id, server_timestamp=time.time(),
+                     presence=presence_dict[user_profile.email])
+        send_event(user_profile.realm, event, active_user_ids(user_profile.realm_id))
```

Comment:
Can this just call `send_presence_changed`?

Reply:
`send_presence_changed` needs a Presence object and not presence dict that we get from get_presence_for_user`.
If we use `UserPresence.objects.get` to get a Presence object, it may create problem when presence object for a particular user profile does not exist and it is evident in tests.
If we want to use `get_or_create` to handle the above case, I am not sure what should be the default values?

Reply:
@timabbott can you give clarification regarding my above comment?

Reply:
@timabbott can you give clarification regarding my above comment?

Reply:
I think let's plan to think through this again once we've merged the big open presence PR.